pkg/admission/admission.go

```diff
@@ -241,6 +267,52 @@ func (a *Admission) mutatePrometheusRules(ar v1.AdmissionReview) *v1.AdmissionRe
 	return reviewResponse
 }
 
+func (a *Admission) validatePrometheus(ar v1.AdmissionReview) *v1.AdmissionResponse {
```

Comment:
Like we do for validating PrometheusRules and AlertmanagerConfig resources, it would be good to avoid duplicating the validation logic in several places otherwise it's going to be harder to maintain in the long run.